### ISSUE TITLE
Why Space component is keeping empty item?

### ISSUE THREAD

**Issue by fkworld** [ant-design/ant-design#30954]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
If there is some empty items in Space, will display a big blank.

`<Space>{"a"}{null}{null}{"b"}</Space>`

I see code: https://github.com/ant-design/ant-design/blob/1d894a22760a9850a48b459fe71570b5ef4a5d88/components/space/index.tsx#L66

### What does the proposed API look like?
`keepEmpty: boolean`

default is true

false will delete the empty items in `Space`

**Comment by PeachScript** [comment]
I cannot reproduce the "big blank", you can checkout this live demo: https://codesandbox.io/s/antd-reproduction-template-forked-1x365

refer: #26389

**Comment by fkworld** [comment]
My bad. It's not an empty item bug.

My code is like:
```
<Space>
  <Space>
    {'a'}
    <Space>{null}</Space>
  </Space>
  {'b'}
</Space>
```
For more clear diff:
```
{/* right */}
<Space>
  {'a'}
  {null}
  {'b'}
</Space>
<br />

{/* right */}
<Space>
  {'a'}
  <Space>{null}</Space>
  {'b'}
</Space>
<br />

{/* wrong */}
<Space>
  <Space>
    {'a'}
    <Space>{null}</Space>
  </Space>
  {'b'}
</Space>
```

**Comment by zernie** [comment]
Sorry, why is this closed? it is still happening in 5.17.4
